### ISSUE TITLE
"DeprecationWarning - unhandled promise" still being triggered

### ISSUE THREAD

**Issue by ramsterdam85** [nicoh88/homebridge-xiaomi-roborock-vacuum#3]
Plugin: 0.1.4
Miio: 0.15.6
Node v8.10
Checked token (iOS backup method)

Debug output is largely similar
```
"[Xiaomi Mi Robot Vaccum 1st Generation] ERORR GetDevice | No correct API answer from xiaomi/roborock
(node:981) UnhandledPromiseRejectionWarning: undefined
(node:981) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:981) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code."
and later down the line:
"[2018-3-25 18:55:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERORR GetDevice | No correct API answer from xiaomi/roborock
[2018-3-25 18:55:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | No vacuum cleaner is discovered.
[2018-3-25 18:56:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner"
```

Any idea what might be wrong?

**Comment by nicoh88** [comment]
**UnhandledPromiseRejectionWarning** fixed in 0.1.4.

Please update:
`npm update -g --unsafe-perm homebridge-xiaomi-roborock-vacuum`

**Comment by ramsterdam85** [comment]
It seems to try to connect but gets stuck in a loop trying to find a port

"2018-3-26 14:35:09] [Xiaomi Mi Robot Vaccum 1st Generation] Initializing XiaomiRoborockVacuum accessory...
[2018-3-26 14:35:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +0ms
  miio:network Making network active, creating socket +2ms
  thing:miio:pending Using manual token: xxxxxxxxxxxxxxxxxxxxx (32 characters) +0ms
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +0ms"

Then this in a loop:
"018-3-26 14:35:39] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +27s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 52818 +4ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:35:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:36:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +0ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 57897 +5ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:36:11] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:36:39] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 35979 +4ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:36:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:37:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 35587 +2ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +0ms
[2018-3-26 14:37:11] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:37:39] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +0ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 35308 +4ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:37:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:38:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +0ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 48790 +3ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +0ms
[2018-3-26 14:38:11] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:38:39] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 52331 +3ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +0ms
[2018-3-26 14:38:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:39:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 53593 +3ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:39:11] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:39:39] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +1ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup)  +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 59740 +2ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +0ms
[2018-3-26 14:39:41] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined
[2018-3-26 14:40:09] [Xiaomi Mi Robot Vaccum 1st Generation] Discovering vacuum cleaner
  miio:network Grabbing reference to network +28s
  miio:network Making network active, creating socket +0ms
  thing:miio:pending Using manual token: xxxx (32 char token- checked in phone backup) +30s
  miio:packet -> <Buffer 21 31 00 20 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff> +30s
  miio:network Network bound to port 38970 +3ms
  miio:network Releasing reference to network +2s
  miio:network Network no longer active, destroying socket +1ms
[2018-3-26 14:40:11] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined

"

**Comment by Nastras** [comment]
With me exactly the error is also after each restart of the homebridge.


Mär 26 18:08:37 HomeBridgeServer homebridge[17650]: (node:17650) UnhandledPromiseRejectionWarning: undefined
Mär 26 18:08:37 HomeBridgeServer homebridge[17650]: (node:17650) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Mär 26 18:08:37 HomeBridgeServer homebridge[17650]: (node:17650) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by ramsterdam85** [comment]
tl;dr for the story above:
0.1.5 works as designed for a short while, but I still get this error
[2018-3-26 16:52:43] [Xiaomi Mi Robot Vaccum 1st Generation] ERROR, Watch | undefined

And then after ~5minutes it goes to "not responding" in the home app.
At some point it seems to crash homebridge, so I had to disable it for the time being.

**Comment by nicoh88** [comment]
Please update to 0.2.0.

`npm update -g --unsafe-perm homebridge-xiaomi-roborock-vacuum`

**An info would be cool, then close. 👍**

**Comment by nicoh88** [comment]
@ramsterdam85  @Nastras 

Fixed "UnhandledPromiseRejectionWarning" with version 0.2.0?
Then i would close this issue.

**Comment by Nastras** [comment]
I would say yes, have not appeared again👌

**Comment by ramsterdam85** [comment]
Works in homebridge without crashing, but setting the "ventilator" to any percentage makes it revert back to "off" in a matter of second. The log goes as follows:
```
[2018-3-29 10:29:59] [Xiaomi Mi Robot Vaccum 1st Generation] ACTION | Start cleaning.
[2018-3-29 10:30:00] [Xiaomi Mi Robot Vaccum 1st Generation] ACTION | FanSpeed set to 90%.
[2018-3-29 11:03:43] [Xiaomi Mi Robot Vaccum 1st Generation] Pause possible true.
[2018-3-29 11:03:43] [Xiaomi Mi Robot Vaccum 1st Generation] Cleaning is false.
[2018-3-29 11:03:43] [Xiaomi Mi Robot Vaccum 1st Generation] Fanspeed is 90%.
[2018-3-29 11:03:43] [Xiaomi Mi Robot Vaccum 1st Generation] Docked is false.

```
Meanwhile on homebridge it goes back to "off" pretty instantly - any idea what may be up?

**Comment by nicoh88** [comment]
@ramsterdam85: Please create an new issue with debug-log. Thank you.

Did you update to version 0.2.1?
